api/v1alpha1/nutanix_clusterconfig_types.go

```diff
@@ -0,0 +1,162 @@
+			Description: "Nutanix Prism Central endpoint configuration",
+			Type:        "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"host": {
+					Description: "host ip/fqdn for Prism Central Server",
+					Type:        "string",
+				},
+				"port": {
+					Description: "port for Prism Central Server",
+					Type:        "integer",
+				},
+				"insecure": {
+					Description: "Prism Central Certificate checking",
+					Type:        "boolean",
+				},
+				"additionalTrustBundle": {
+					Description: "Name of configMap with certificate trust bundle used for Prism Central connection",
+					Type:        "string",
+				},
+				"credentialSecret": {
+					Description: "Name of a Credential information secret for the target Prism instance",
+					Type:        "string",
+				},
+			},
+		},
+	}
+}
+
+type NutanixControlPlaneEndpointSpec struct {
+	Host string `json:"host,omitempty"`
+	Port int32  `json:"port,omitempty"`
+}
+
+func (NutanixControlPlaneEndpointSpec) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Description: "Nutanix control-plane endpoint configuration",
+			Type:        "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"host": {
+					Description: "host ip/fqdn for control plane API Server",
+					Type:        "string",
+				},
+				"port": {
+					Description: "port for control plane API Server",
+					Type:        "integer",
+				},
+			},
+		},
+	}
+}
+
+type NutanixFailureDomains []NutanixFailureDomain
+
+func (NutanixFailureDomains) VariableSchema() clusterv1.VariableSchema {
+	resourceSchema := NutanixFailureDomain{}.VariableSchema().OpenAPIV3Schema
+
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Description: "Nutanix failure domains",
+			Type:        "array",
+			Items:       &resourceSchema,
+		},
+	}
+}
+
+type NutanixFailureDomain struct {
+	// name defines the unique name of a failure domain.
+	// Name is required and must be at most 64 characters in length.
+	// It must consist of only lower case alphanumeric characters and hyphens (-).
+	// It must start and end with an alphanumeric character.
+	// This value is arbitrary and is used to identify the failure domain within the platform.
+	Name string `json:"name"`
+
+	// cluster is to identify the cluster (the Prism Element under management of the Prism Central),
+	// in which the Machine's VM will be created. The cluster identifier (uuid or name) can be obtained
+	// from the Prism Central console or using the prism_central API.
+	Cluster NutanixResourceIdentifier `json:"cluster"`
+
+	// subnets holds a list of identifiers (one or more) of the cluster's network subnets
+	// for the Machine's VM to connect to. The subnet identifiers (uuid or name) can be
+	// obtained from the Prism Central console or using the prism_central API.
+	Subnets []NutanixResourceIdentifier `json:"subnets"`
```

Comment:
currently facing following issue in capi while applying patches <pre>
E0329 21:46:37.778027       1 reconcile_state.go:1113] "Failed to create object" err="NutanixMachineTemplate.infrastructure.cluster.x-k8s.io \"nutanix-quick-start-helm-addon-cilium-npscb\" is invalid: spec.template.spec.subnet[0].type: Unsupported value: \"\": supported values: \"uuid\", \"name\"" controller="topology/cluster" controllerGroup="cluster.x-k8s.io" controllerKind="Cluster" Cluster="default/nutanix-quick-start-helm-addon-cilium" namespace="default" name="nutanix-quick-start-helm-addon-cilium" reconcileID="ded8efe6-16c4-4f6e-a6c7-35c8f6e231c5" resource={"group":"infrastructure.cluster.x-k8s.io","version":"v1beta1","resource":"NutanixMachineTemplate"} NutanixMachineTemplate="default/nutanix-quick-start-helm-addon-cilium-npscb" NutanixMachineTemplate="default/nutanix-quick-start-helm-addon-cilium-npscb"
E0329 21:46:37.779396       1 controller.go:329] "Reconciler error" err="error reconciling the Cluster topology: failed to create NutanixMachineTemplate.infrastructure.cluster.x-k8s.io: FieldValueNotSupported: spec.template.spec.subnet[0].type: Unsupported value: \"\": supported values: \"uuid\", \"name\"" controller="topology/cluster" controllerGroup="cluster.x-k8s.io" controllerKind="Cluster" Cluster="default/nutanix-quick-start-helm-addon-cilium" namespace="default" name="nutanix-quick-start-helm-addon-cilium" reconcileID="ded8efe6-16c4-4f6e-a6c7-35c8f6e231c5"
</pre>